src/App.jsx

```diff
@@ -14,45 +15,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
+
+export const App = () => {
+  let goods = [...goodsFromServer];
+  const [sortField, setSortField] = useState('');
+  const [reverseList, setReverseList] = useState(false);
+
+  switch (sortField) {
+    case SORT_FIELD_ALPHABETICALLY:
+      goods = goods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SORT_FIELD_LENGTH:
+      goods = goods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverseList) {
+    goods = goods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortField !== SORT_FIELD_ALPHABETICALLY ? 'is-light' : ''}`}
+          onClick={() => (
+            setSortField(SORT_FIELD_ALPHABETICALLY)
+          )}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success ${sortField !== SORT_FIELD_LENGTH ? 'is-light' : ''}`}
+          onClick={() => (
+            setSortField(SORT_FIELD_LENGTH)
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${reverseList ? '' : 'is-light'}`}
+          onClick={() => {
```

Comment:
Move the onClick logic to a separate function. It follows the best practice of separating concerns and keeping your JSX code clean and focused on rendering components rather than defining complex logic directly within it.

Reply:
Not fixed 
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/99403198/e0cac84c-bb36-426f-b908-105870605d98)